packages/dds/sequence/src/test/revertibles.spec.ts

```diff
@@ -0,0 +1,508 @@
+		collection2 = sharedString2.getIntervalCollection("test");
+	});
+
+	it("remote string remove interacting with reverting an interval remove", () => {
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.SlideOnRemove).getIntervalId()!;
+		collection.removeIntervalById(id);
+		containerRuntimeFactory.processAllMessages();
+
+		sharedString2.removeRange(1, 3);
+		containerRuntimeFactory.processAllMessages();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assert.equal(sharedString.getText(), sharedString2.getText());
+		assertIntervals(sharedString, collection, [{ start: 0, end: 3 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 3 }]);
+	});
+	it("remote string remove that shares an endpoint with a removed interval that gets reverted", () => {
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.SlideOnRemove).getIntervalId()!;
+		collection.removeIntervalById(id);
+		containerRuntimeFactory.processAllMessages();
+
+		sharedString2.removeRange(0, 3);
+		containerRuntimeFactory.processAllMessages();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assert.equal(sharedString.getText(), sharedString2.getText());
+		assertIntervals(sharedString, collection, [{ start: 0, end: 2 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 2 }]);
+	});
+	it("remote string add interacting with reverting an interval remove", () => {
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 7, IntervalType.SlideOnRemove).getIntervalId()!;
+		containerRuntimeFactory.processAllMessages();
+
+		collection.removeIntervalById(id);
+		sharedString2.insertText(5, " hi");
+		containerRuntimeFactory.processAllMessages();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assert.equal(sharedString.getText(), sharedString2.getText());
+		assertIntervals(sharedString, collection, [{ start: 0, end: 10 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 10 }]);
+	});
+	it("remote interval change interacting with reverting an interval remove", () => {
```

Comment:
this test still fails with the 0x2f5 assertion, even after updating revertLocalDelete to use the setSlideOnRemove code to set the type correctly. Is there something else I have to do, or anything else that could cause this problem? @Abe27342

Reply:
The problem is that the remote change is being applied over the locally re-created interval that is not yet acked, so the endpoints are unexpectedly stayOnRemove. Not sure what the right fix is. We might have to drop remote edits to pending re-creates. Maybe re-create with a new id if it's not valid to hang on to ids across undo-redo.

Reply:
recreating with a new id seems like a safe fix. *maybe* we want to support stable id across undo-redo at some point, but i'm not sure anyone cares right now.

Reply:
Uh oh: if the id changes, prior revertibles won't find the interval anymore. E.g. change an interval, delete it, revert the delete, then try to revert the change -> change revert no-ops because it doesn't find the interval.
The revertibles hold on the intervals themselves rather than the id, so I think it should work if `revertLocalDelete` updates the id on the interval in the revertible. (Also add a test for this.)